#### This is a script to check how similar are the training and the dev sets

In [1]:
#imports 
import pandas as pd 
import csv 
import re 
from pathlib import Path 
from tqdm import tqdm 

<hr style="border:1.0px solid gray"></hr> 

#### Define in/out files 
    
<hr style="border:1.0px solid gray"></hr> 

In [19]:
dgs_dir = Path('path')

glosses_stem_train = dgs_dir / 'glosses_stem_train.txt'
glosses_stem_dev = dgs_dir / 'glosses_stem_dev.txt'

glosses_train = dgs_dir / 'glosses_train.txt'
glosses_dev = dgs_dir / 'glosses_dev.txt' 

dgs_stripped_train = dgs_dir / 'german_stripped_train.txt' 
dgs_stripped_dev = dgs_dir / 'german_stripped_dev.txt'

###############################################################

phoenix_dir = Path('path')

phoenix_train = phoenix_dir / 'phoenix_train_glosses.txt'
phoenix_dev = phoenix_dir / 'phoenix_dev_glosses.txt' 

phoenix_sentences_train = phoenix_dir / 'phoenix_train_sentences.txt' 
phoenix_sentences_dev = phoenix_dir / 'phoenix_dev_sentences.txt' 

###############################################################

assert glosses_stem_train.exists(), 'dgs training set not found'
assert glosses_stem_dev.exists(), 'dgs dev set not found'
assert phoenix_train.exists(), 'phoenix training set not found'
assert phoenix_dev.exists(), 'phoenix dev set not found' 

In [6]:
line = "This is a sentence"

In [7]:
line.split()

['This', 'is', 'a', 'sentence']

In [9]:
def make_gloss_set(in_txt):
    glosses_df = pd.read_csv(in_txt, header=None)
    list_of_gloss_sentences = glosses_df[0].tolist()
    
    list_of_glosses = []
    
    for sentence in list_of_gloss_sentences:
        for gloss in str(sentence).split():
            list_of_glosses.append(gloss)
    
    len_list = len(list_of_glosses)
    set_of_glosses = set(list_of_glosses)
    len_set = len(set_of_glosses)
    return(set_of_glosses, len_list, len_set) 

#### Phoenix training set has 55 247 glosses and 1085 of them are unique. 

#### Phoenix dev set has 3 748 glosses and 393 of them are unique.

#### Overlap of the training and dev set of Phoenix is 381 glosses.

---

#### DGS training set has 301 772 glosses and 4 559 of them are unique.

#### DGS dev set has 21 715 glosses and 1909 of them are unique.

#### Overlap of training and dev set of the stemmed DGS glosses is 1845 glosses.

---

#### DGS original training set has 301 772 glosses and 19 521 of them are unique.

#### DGS original dev set has 21 715 glosses and 4 891 of them are unique.

In [52]:
dgs_original_overlap = list(make_gloss_set(glosses_train)[0] & make_gloss_set(glosses_dev)[0])  

In [54]:
len(dgs_original_overlap)

4418

#### Overlap of original training and dev set of DGS is 4418 glosses.